homeassistant/components/google_air_quality/config_flow.py

```diff
@@ -0,0 +1,195 @@
+            return self.async_abort(
+                reason="access_not_configured",
+                description_placeholders={"message": str(ex)},
+            )
+        except Exception:
+            self.logger.exception("Unknown error occurred")
+            return self.async_abort(reason="unknown")
+        await self.async_set_unique_id(user_resource_info.id)
+        self._abort_if_unique_id_configured()
+        return self.async_create_entry(
+            title=user_resource_info.name,
+            data=data,
+        )
+
+
+class LocationSubentryFlowHandler(ConfigSubentryFlow):
+    """Handle subentry flow for adding and modifying a location."""
+
+    async def async_step_user(
+        self,
+        user_input: dict[str, Any] | None = None,
+    ) -> SubentryFlowResult:
+        """User flow to add a new location."""
+        if user_input is not None:
+            entry = self._get_entry()
+            hass = self.hass
+            implementation = (
+                await config_entry_oauth2_flow.async_get_config_entry_implementation(
+                    hass, entry
+                )
+            )
+            web_session = async_get_clientsession(hass)
+            oauth_session = config_entry_oauth2_flow.OAuth2Session(
+                hass, entry, implementation
+            )
+            auth = api.AsyncConfigEntryAuth(web_session, oauth_session)
+            client = GoogleAirQualityApi(auth)
+            location = user_input[CONF_LOCATION]
+            lat = location[CONF_LATITUDE]
+            lon = location[CONF_LONGITUDE]
+            try:
+                await client.async_air_quality(lat, lon)
+            except NoDataForLocationError:
+                return self._show_form_user(
+                    user_input,
+                    errors={"base": "no_data_for_location"},
+                )
+            except GoogleAirQualityApiError as ex:
+                return self.async_abort(
+                    reason="access_not_configured",
+                    description_placeholders={"message": str(ex)},
+                )
+            except Exception:
+                _LOGGER.exception("Unknown error occurred")
+                return self.async_abort(reason="unknown")
+
+            unique_id = f"{lat}_{lon}"
+            for entry in hass.config_entries.async_entries(DOMAIN):
+                for subentry in entry.subentries.values():
+                    if subentry.unique_id == unique_id:
+                        return self.async_abort(reason="already_configured")
+            data = {
+                CONF_LATITUDE: lat,
+                CONF_LONGITUDE: lon,
+            }
+            try:
+                geo_data = await client.async_reverse_geocode(lat, lon)
+                title = geo_data.results[0].formatted_address
+            except (GoogleAirQualityApiError, ValueError, IndexError):
+                _LOGGER.exception("Could not resolve address for %s,%s:", lat, lon)
```

Comment:
```suggestion
                _LOGGER.error("Could not resolve address for %s,%s:", lat, lon)
```
this is a known error, so we shouldn't need the stacktrace, or do we